modules/mobile/spec/models/adapters/appeals_adapter_spec.rb

```diff
@@ -0,0 +1,125 @@
+                    { 'date' => '2015-07-24', 'type' => 'bva_decision' },
+                    { 'date' => '2015-10-06', 'type' => 'ssoc' },
+                    { 'date' => '2016-05-03', 'type' => 'bva_decision' },
+                    { 'date' => '2018-01-16', 'type' => 'ssoc' }],
+                  'evidence' => [],
+                  'incompleteHistory' => false,
+                  'issues' =>
+                   [{ 'active' => true, 'date' => '2016-05-03',
+                      'description' => 'Increased rating, migraines',
+                      'diagnosticCode' => '8100', 'lastAction' => 'remand' },
+                    { 'active' => true, 'date' => '2016-05-03',
+                      'description' => 'Increased rating, limitation of leg motion',
+                      'diagnosticCode' => '5260', 'lastAction' => 'remand' },
+                    { 'active' => true, 'date' => '2016-05-03',
+                      'description' => '100% rating for individual unemployability',
+                      'diagnosticCode' => nil, 'lastAction' => 'remand' },
+                    { 'active' => false, 'date' => nil, 'description' => 'Service connection, ankylosis of hip',
+                      'diagnosticCode' => '5250', 'lastAction' => nil },
+                    { 'active' => true, 'date' => '2015-07-24',
+                      'description' => 'Service connection, degenerative spinal arthritis', 'diagnosticCode' => '5242',
+                      'lastAction' => 'remand' },
+                    { 'active' => false, 'date' => nil, 'description' => 'Service connection, hearing loss',
+                      'diagnosticCode' => '6100', 'lastAction' => nil },
+                    { 'active' => true, 'date' => '2015-07-24',
+                      'description' => 'Service connection, sciatic nerve paralysis', 'diagnosticCode' => '8520',
+                      'lastAction' => 'remand' },
+                    { 'active' => false, 'date' => nil, 'description' => 'Service connection, arthritis due to trauma',
+                      'diagnosticCode' => '5010', 'lastAction' => nil },
+                    { 'active' => false, 'date' => '2015-07-24',
+                      'description' =>
+                        'New and material evidence for service connection, degenerative spinal arthritis',
+                      'diagnosticCode' => '5242',
+                      'lastAction' => 'allowed' }],
+                  'location' => 'aoj',
+                  'programArea' => 'compensation',
+                  'status' => { 'details' => {}, 'type' => 'remand_ssoc' },
+                  'type' => 'post_remand',
+                  'updated' => '2018-01-19T10:20:42-05:00' },
+       'id' => '3294289',
+       'type' => 'legacyAppeal' }]
+  end
+
+  def appeal_by_id(id, overrides: {}, without: [])
+    appeal = appeals.find { |a| a['id'] == id }
+    appeal['attributes'].merge!(overrides.stringify_keys) if overrides.any?
```

Comment:
😉 Looks like you were right that this would come in handy again. I guess if we do it one more time we'll have to extract it into a helper method.

Reply:
yeah, i also realized how unique it had to be to how the upstream response is structured so making it generic is possible but a bit more complex.